langkit/templates/ocaml_api/struct_types_ocaml.mako

```diff
@@ -23,7 +23,15 @@ end
       % if f.type.is_ast_node:
     ${ocaml_api.field_name(f)} : ${ocaml_api.c_value_type(f.type, cls)};
       % else:
+         % if f.type.is_entity_type:
+            ## If the type is an entity type, we want an optional type instead.
+            ## This is because we have no way to know if the node will be null
+            ## or not.
+    ${ocaml_api.field_name(f)} :
+      ${ocaml_api.type_public_name(f.type, cls)} option;
+         % else:
     ${ocaml_api.field_name(f)} : ${ocaml_api.type_public_name(f.type, cls)};
+         % endif
```

Comment:
There is a lot of code duplication, here: may I suggest:

```ocaml
${ocaml_api.field_name(f)} :
  % if f.type.is_ast_node:
    ${ocaml_api.c_value_type(f.type, cls)}
  % elif f.type.is_entity_type:
    ## If the type is an entity type, we want an optional type instead.
    ## This is because we have no way to know if the node will be null
    ## or not.
    ${ocaml_api.type_public_name(f.type, cls)} option
  % else:
    ${ocaml_api.type_public_name(f.type, cls)};
  % endif
;
```

Reply:
Given that OCaml is not a indentation-sensitive language, I think it’s more readable for templates to be written so that indentation for the Mako syntax prevails over indentation for the OCaml code: here it’s hard to understand the structure of the code looking at the indentation, whereas conceptually there are if/elif/else controls in which all clauses emit a type expression. What do you think?